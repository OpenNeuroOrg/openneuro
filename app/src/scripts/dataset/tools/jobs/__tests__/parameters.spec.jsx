import React from 'react'
import { shallow } from 'enzyme'
import JobParameters from '../parameters'

describe('dataset/tools/jobs/parameters', () => {
  const onChange = jest.fn()
  const onRestoreDefaults = jest.fn()
  const params = {
    boolean: true,
    participant_label: '01,02',
    another_option: 'some_string_value',
  }

  let options = ['option1', 'option2', 'option3']
  let defChecked = ['option1']

  const metadata = {
    boolean: {
      type: 'checkbox',
      required: false,
      description: 'some kind of checkbox',
      options: options,
      defaultChecked: defChecked,
    },
    participant_label: {
      type: 'select',
      required: false,
      description: 'job participants',
    },
    another_option: {
      type: 'text',
      required: true,
      hidden: true,
      defaultValue: 'some default value',
      description: 'some kind of text option',
    },
    radio_option: {
      type: 'radio',
      required: false,
      description: 'some kind of radio option',
      options: options,
    },
    list_option: {
      type: 'select',
      required: false,
      description: 'some kind of select list',
      options: options,
    },
  }

  const subjects = ['01', '02']
  const arrInput = ['check1', 'check2']

  it('renders successfully', () => {
    const wrapper = shallow(
      <JobParameters
        parameters={params}
        parametersMetadata={metadata}
        subjects={subjects}
        arrInput={arrInput}
        onChange={onChange}
        onRestoreDefaults={onRestoreDefaults}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('renders noscript with no parameters', () => {
    const wrapper = shallow(<JobParameters />)
    expect(wrapper.first().type()).toBe('noscript')
  })
})
