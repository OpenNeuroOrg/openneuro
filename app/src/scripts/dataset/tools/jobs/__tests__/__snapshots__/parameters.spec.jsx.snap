// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataset/tools/jobs/parameters renders successfully 1`] = `
<div>
  <br />
  <hr />
  <br />
  <div
    className="row"
  >
    <div
      className="col-xs-6"
    >
      <h5>
        Parameters
      </h5>
    </div>
    <div
      className="col-xs-6 default-reset"
    >
      <button
        className="btn-reset"
        onClick={[MockFunction]}
      >
        Restore Default Parameters
      </button>
    </div>
  </div>
  <div
    className={null}
    id={null}
    key="boolean"
  >
    <div
      className="parameters form-horizontal"
    >
      <div
        className="form-group"
        key="boolean"
      >
        <label
          className="sr-only"
        >
          boolean
        </label>
        <div
          className="input-group"
        >
          <div
            className="input-group-addon"
          >
            boolean
          </div>
          <div
            className="clearfix"
          >
            <CheckOrRadio
              controlFunc={[Function]}
              defaultChecked={
                Array [
                  "option1",
                ]
              }
              setName="boolean"
              type="checkbox"
            />
            <span
              className="help-text"
            >
              some kind of checkbox
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className={null}
    id={null}
    key="participant_label"
  >
    <div
      className="parameters form-horizontal"
    >
      <div
        className="form-group"
        key="participant_label"
      >
        <label
          className="sr-only"
        >
          participant_label
        </label>
        <div
          className="input-group"
        >
          <div
            className="input-group-addon"
          >
            participant_label
          </div>
          <div
            className="clearfix"
          >
            <Select
              arrowRenderer={[Function]}
              autosize={true}
              backspaceRemoves={true}
              backspaceToRemoveMessage="Press backspace to remove {label}"
              clearAllText="Clear all"
              clearRenderer={[Function]}
              clearValueText="Clear value"
              clearable={true}
              closeOnSelect={true}
              deleteRemoves={true}
              delimiter=","
              disabled={false}
              escapeClearsValue={true}
              filterOptions={[Function]}
              ignoreAccents={true}
              ignoreCase={true}
              inputProps={Object {}}
              isLoading={false}
              joinValues={false}
              labelKey="label"
              matchPos="any"
              matchProp="any"
              menuBuffer={0}
              menuRenderer={[Function]}
              multi={true}
              noResultsText="No results found"
              onBlurResetsInput={true}
              onChange={[Function]}
              onCloseResetsInput={true}
              onSelectResetsInput={true}
              openOnClick={true}
              optionComponent={[Function]}
              options={
                Array [
                  "01",
                  "02",
                ]
              }
              pageSize={5}
              placeholder="Select your subject(s)"
              removeSelected={true}
              required={false}
              rtl={false}
              scrollMenuIntoView={true}
              searchable={true}
              simpleValue={true}
              tabSelectsValue={true}
              trimFilter={true}
              value="01,02"
              valueComponent={[Function]}
              valueKey="value"
            />
            <span
              className="help-text"
            >
              job participants
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="required-param"
    id="hidden"
    key="another_option"
  >
    <div
      className="parameters form-horizontal"
    >
      <div
        className="form-group"
        key="another_option"
      >
        <label
          className="sr-only"
        >
          another_option
        </label>
        <div
          className="input-group"
        >
          <div
            className="input-group-addon"
          >
            another_option
          </div>
          <div
            className="clearfix"
          >
            <input
              className="form-control"
              onChange={[Function]}
              value="some_string_value"
            />
            <span
              className="help-text"
            >
              some kind of text option
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
